docs/deprecation.md

```diff
@@ -39,5 +39,60 @@ In order to deprecate a feature gate the following conditions must be met:
 
 After the above deprecation process, the feature gate should remain as a no-op with a warning popping up.
 
+### Metadata (e.g. labels / annotations)
+Kubevirt adds certain labels / annotations to both objects that are created by us (e.g. VMs) and cluster-wide
+objects (e.g. Nodes). This metadata is exposed to the end-user which might use it and depend on it.
+
+Usually, metadata needs to be deprecated if it has a non-intuitive / irrelevant name or reflects data that is no
+longer relevant / useful.
+
+#### Deprecation
+In order to deprecate metadata the following conditions must be met:
+* A mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform about the deprecation.
+* In case the metadata is being renamed / changed, the new metadata (e.g. label) should be added alongside the deprecated
+one.
+
+For example, if we want to deprecate a label named `kubevirt.io/badNonIntuitiveName` to
+`kubevirt.io/greatNewName`, both labels should appear.
+
+#### Removal
```

Comment:
We can only remove elemts from the API by increasing the major version of the PAI.

IOW- We follow semver, semver requires api compatibility within a major version of an API.
We must follow this.

Kube also picks this up in https://kubernetes.io/docs/reference/using-api/deprecation-policy/ with rule https://github.com/kubevirt/kubevirt/pull/1

Reply:
Thank you! A great point indeed.
Let's move the discussion to https://github.com/kubevirt/kubevirt/pull/9012#discussion_r1229514274

Reply:
+1

Reply:
> We can only remove elemts from the API by increasing the major version of the PAI.
> 
> IOW- We follow semver, semver requires api compatibility within a major version of an API. We must follow this.
> 
> Kube also picks this up in https://kubernetes.io/docs/reference/using-api/deprecation-policy/ with rule No. 1

I'll reply in the main thread above but I don't think the k8s deprecation policy suggests moving between major versions when removing elements as your suggesting here. Just that you increment the overall version. IOW dropping an element from `v1beta1` would require `v1beta2` or `v1`.

Reply:
I do not think we can follow K8S rules, unless I understand them wrongly.
If we add a field to the VM spec, initially for a feature that is in alpha (not that we follow such a notion), the rule mentioned here will not allow us to revert or change anything.

I'm either not clearly grasping something basic or I do not understand how this can work.